databricks/koalas/indexes.py

```diff
@@ -1313,205 +1314,61 @@ def levshape(self):
-            # for string type basically, but `kidx3` returns `True` since pandas returns `True`
-            # if all values belong to same index partition('y', this example) are `None`.
-            # so, below lines to handle above cases.
-            cond = ((col >= F.lag(col, 1).over(window)) &
-                    F.when(col.isNull(), F.lag(col, 1).over(window).isNull())
-                     .otherwise(F.lag(col, 1, True).over(window).isNotNull()))
-
-        internal = _InternalFrame(
-            sdf=sdf.select(cond.alias(index_name)),
-            index_map=[(index_name, None)])
-
-        # first of all, make sure if the first index is monotonic increasing
-        # if first index is not monotonic increasing, return False
-        if not DataFrame(internal).index.all():
-            return False
-
-        def _is_nlevel_monotonic(level):
-            """
-            Return if given level of index is monotonic.
-            (`level` should be > 0)
-            """
-            col = index_scols[level]
-            col_prev = index_scols[level - 1]
-
-            # each level of index should compare their values based on grouped previous index.
-            window = (Window.orderBy(NATURAL_ORDER_COLUMN_NAME)
-                            .partitionBy(col_prev)
-                            .rowsBetween(-1, -1))
-
-            if sdf.dtypes[level][1] in ['bigint', 'double']:
-                cond = col >= F.lag(F.when(col.isNotNull(), col).otherwise(np.inf), 1).over(window)
+    def _is_monotonic(self):
+        col = self._scol
+        window = Window.orderBy(NATURAL_ORDER_COLUMN_NAME).rowsBetween(-1, -1)
+        prev = F.lag(col, 1).over(window)
+
+        cond = F.lit(True)
+        for field in self.spark_type[::-1]:
+            left = col.getField(field.name)
+            right = prev.getField(field.name)
+            if isinstance(field.dataType, StringType):
+                compare = compare_disallow_null
+            elif isinstance(field.dataType, BooleanType):
+                compare = compare_allow_null
+            elif isinstance(field.dataType, NumericType):
+                compare = compare_null_first
             else:
-                cond = ((col >= F.lag(col, 1).over(window)) &
-                        F.when(col.isNull(), F.lag(col, 1).over(window).isNull())
-                         .otherwise(F.lag(col, 1, True).over(window).isNotNull()))
-
-            internal = _InternalFrame(
-                sdf=sdf.select(cond.alias(index_name)),
-                index_map=[(index_name, None)])
-            return DataFrame(internal).index.all()
-
-        # we already checked the first index (level = 0) above,
-        # so, start checking from level 1.
-        for nlevel in range(self.nlevels)[1:]:
-            if not _is_nlevel_monotonic(nlevel):
-                return False
-
-        return True
+                compare = compare_null_last
```

Comment:
Need to check more.

Reply:
Thanks for the suggestion and sorry for check late. 😅 

okay let me test this more after merging this for now.

and the other codes all looks good to me :)